lightclient/src/background.rs

```diff
@@ -43,59 +47,58 @@ pub enum FromSubxt {
         params: String,
         /// Channel used to send back the subscription ID if successful.
         sub_id: oneshot::Sender<MethodResponse>,
-        /// Channel used to send back the notifcations.
+        /// Channel used to send back the notifications.
         sender: mpsc::UnboundedSender<Box<RawValue>>,
+        /// The ID of the chain used to identify the chain.
+        chain_id: ChainId,
     },
 }
 
 /// Background task data.
-pub struct BackgroundTask {
+pub struct BackgroundTask<TPlatform: PlatformRef, TChain> {
     /// Smoldot light client implementation that leverages the exposed platform.
-    client: smoldot_light::Client<PlatformType>,
-    /// The ID of the chain used to identify the chain protocol (ie. substrate).
-    ///
-    /// Note: A single chain is supported for a client. This aligns with the subxt's
-    /// vision of the Client.
-    chain_id: ChainId,
-    /// Unique ID for RPC calls.
-    request_id: usize,
+    client: smoldot_light::Client<TPlatform, TChain>,
+    /// Unique IDs for RPC calls.
+    rpc_chain_id: UniqueChainID,
```

Comment:
I find the explanation and the naming hard to understand.
My suggestion is to name this `request_id_per_chain` or something similar

However, I don't understand why the request IDs has been groped by chain because it will just be incremented on every RPC call, no?

Reply:
Yep indeed, for RPC calls we control the request ID and the smoldot / node needs to provide the same ID in the response. Initially I avoided at all costs adding another dimension (chainID) to the internal mapping. 

However, for subscriptions we need to make one plain RPC method call, and the smoldot / node is free to give us any number in response to identify a subscription ID. 
And since that code was already there, seemed like an easy thing to reuse the logic and have unique IDs wrt to calling different chains. But that could be reverted if you think simplifies things :D

Reply:
Ok, my point is that HashMap could be removed but what you did it's quite clean anyway so I don't mind it.

I just want some better docs here.